core/corehttp/gateway_handler.go

```diff
@@ -257,40 +330,43 @@ func (i *gatewayHandler) getOrHeadHandler(w http.ResponseWriter, r *http.Request
 		return
 	}
 
-	idx, err := i.api.Unixfs().Get(r.Context(), ipath.Join(resolvedPath, "index.html"))
-	switch err.(type) {
-	case nil:
-		dirwithoutslash := urlPath[len(urlPath)-1] != '/'
-		goget := r.URL.Query().Get("go-get") == "1"
-		if dirwithoutslash && !goget {
-			// See comment above where originalUrlPath is declared.
-			http.Redirect(w, r, originalUrlPath+"/", 302)
-			return
-		}
+	if !isReedSolomonSubdirOrFile {
+		idx, err := i.api.Unixfs().Get(r.Context(), ipath.Join(resolvedPath, "index.html"))
+		switch err.(type) {
+		case nil:
+			dirwithoutslash := urlPath[len(urlPath)-1] != '/'
+			goget := r.URL.Query().Get("go-get") == "1"
+			if dirwithoutslash && !goget {
+				// See comment above where originalUrlPath is declared.
+				http.Redirect(w, r, originalUrlPath+"/", 302)
+				return
+			}
 
-		f, ok := idx.(files.File)
-		if !ok {
-			internalWebError(w, files.ErrNotReader)
+			f, ok := idx.(files.File)
+			if !ok {
+				internalWebError(w, files.ErrNotReader)
+				return
+			}
+
+			// write to request
+			http.ServeContent(w, r, "index.html", modtime, f)
+			return
+		case resolver.ErrNoLink:
+			// no index.html; noop
+		default:
+			internalWebError(w, err)
 			return
 		}
-
-		// write to request
-		http.ServeContent(w, r, "index.html", modtime, f)
-		return
-	case resolver.ErrNoLink:
-		// no index.html; noop
-	default:
-		internalWebError(w, err)
-		return
 	}
 
 	if r.Method == "HEAD" {
 		return
 	}
 
-	// storage for directory listing
+	// storage for directory list
 	var dirListing []directoryItem
 	dirit := dir.Entries()
+	dirit.BreadthFirstTraversal()
```

Comment:
Is it safe to do this for non reed solomon folder?

Reply:
It is no-op for non Reed Solomon folder.